addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -58,22 +107,41 @@ public SomfyTahomaItemDiscoveryService(SomfyTahomaBridgeHandler bridgeHandler) {
                 THING_TYPE_EXTERIORVENETIANBLIND, THING_TYPE_GARAGEDOOR, THING_TYPE_ACTIONGROUP, THING_TYPE_AWNING,
                 THING_TYPE_ONOFF, THING_TYPE_LIGHT, THING_TYPE_LIGHTSENSOR, THING_TYPE_SMOKESENSOR,
                 THING_TYPE_CONTACTSENSOR, THING_TYPE_OCCUPANCYSENSOR, THING_TYPE_WINDOW, THING_TYPE_EXTERNAL_ALARM,
-                THING_TYPE_INTERNAL_ALARM, THING_TYPE_POD));
+                THING_TYPE_INTERNAL_ALARM, THING_TYPE_POD, THING_TYPE_HEATING_SYSTEM, THING_TYPE_ONOFF_HEATING_SYSTEM, THING_TYPE_DOOR_LOCK));
     }
 
     @Override
-    public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServiceCallback) {
-        this.discoveryServiceCallback = discoveryServiceCallback;
+    protected void startScan() {
+        runDiscovery();
     }
 
-    @Override
-    protected void startScan() {
-        logger.debug("Starting scanning for items...");
-        bridge.setDiscoveryService(this);
-        bridge.startDiscovery();
+    private synchronized void runDiscovery() {
+        logger.debug("Starting scanning for things...");
+
+        if (bridge.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            SomfyTahomaSetup devices = bridge.listDevices();
+            for (SomfyTahomaDevice device : devices.getDevices()) {
+                discoverDevice(device);
+            }
+            for (SomfyTahomaGateway gw : devices.getGateways()) {
+                gatewayDiscovered(gw.getGatewayId());
+            }
+
+            ArrayList<SomfyTahomaActionGroup> actions = bridge.listActionGroups();
+
+            for (SomfyTahomaActionGroup group : actions) {
+                String oid = group.getOid();
+                String label = group.getLabel();
+
+                //actiongroups use oid as deviceURL
+                actionGroupDiscovered(label, oid, oid);
+            }
+        } else {
+            logger.warn("Cannot start discovery since the bridge is not online!");
```

Comment:
Probably not very useful for the user since he sees that his thing is offline and thus will not expect anything from it. So please reduce to debug.

Reply:
Changed